File: shanoir-ng-front/src/app/carmin/models/pipelineParameter.ts
/**
 * Shanoir NG - Import, manage and share neuroimaging data
 * Copyright (C) 2009-2022 Inria - https://www.inria.fr/
 * Contact us on https://project.inria.fr/shanoir/
 * 
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 * 
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see https://www.gnu.org/licenses/gpl-3.0.html
 */
import { ParameterType } from './parameterType';

export interface PipelineParameter { 
    name: string;
    type: ParameterType;
    isOptional: boolean;
    isReturnedValue: boolean;
    /**
     * Default value. It must be consistent with the parameter type.
     */
    defaultValue?: any; // @alaeessaki not described in swagger in java client, it's described as an Object, but i think to avoid errors for now i'll make it any. TODO specify the tyoe.
    description?: string;
}

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/vip/monitoring/model/PipelineParameter.java
/**

 * Shanoir NG - Import, manage and share neuroimaging data
 * Copyright (C) 2009-2019 Inria - https://www.inria.fr/
 * Contact us on https://project.inria.fr/shanoir/
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see https://www.gnu.org/licenses/gpl-3.0.html
 */

package org.shanoir.ng.vip.monitoring.model;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;

/**
 * This class represents the parameter of a VIP pipeline.
 */
@Entity
public class PipelineParameter {

    @Id
    private Long id;

    private String name;
    private String type;
    private String value;
    private boolean isOptional;
    private boolean isReturnedValue;
    private String description;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public boolean isOptional() {
        return isOptional;
    }

    public void setOptional(boolean optional) {
        isOptional = optional;
    }

    public boolean isReturnedValue() {
        return isReturnedValue;
    }

    public void setReturnedValue(boolean returnedValue) {
        isReturnedValue = returnedValue;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getId() {
        return id;
    }
}
